### PR TITLE
Mark pyldap as optional

### LINKED ISSUE #457: Mark ldap as optional dependency
Brought up in #409...

`ldap` can be marked as an optional dependency in `setup.py`. The imports should be surrounded by `try` and throw `ImproperlyConfigured` if `ldap` is not installed.

Test requiring `ldap` should be skipped if `ldap` is not installed.

### PR DESCRIPTION
**Pull request by tpazderka**
Close #457

- [x] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---

### CODE REVIEW

File: src/oic/utils/authn/ldapc.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
-import ldap
+try:
+    import ldap
+except ImportError:
+    raise ImportError('This module can be used only with ldap isntalled.')
```
Comment (lwm) [inline src/oic/utils/authn/ldapc.py]: isntalled -> installed. Also, should we here instead mention the package - pyldap? Might be faster for the user.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,10 @@ The format is based on the [KeepAChangeLog] project.
 ### Added
 - [#443] Ability to specify additional supported claims for oic.Provider
 
+### Changed
+- [#457] ldap is now an optional dependency. ``oic.utils.authn.ldapc`` and ``oic.utils.userinfo.ldap_info`` raise
+         ``ImportError`` on import if ``ldap`` is not present
```
Comment (lwm) [inline CHANGELOG.md]: Same - should we instead mention pyldap instead of ldap?

File: src/oic/utils/userinfo/ldap_info.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
-import logging
+try:
+    import ldap
+except ImportError:
+    raise ImportError('This module can be used only with ldap installed.')
```
Comment (lwm) [inline src/oic/utils/userinfo/ldap_info.py]: ^^^

**Comment by lwm** [comment]
👍

**Comment by tpazderka** [comment]
Fixed.

**Pull request merged**